c/tskit/tables.c

```diff
@@ -94,22 +97,32 @@ read_table_cols(kastore_t *store, read_table_col_t *read_cols, size_t num_cols)
         *read_cols[j].len_dest = (tsk_size_t) -1;
     }
     for (j = 0; j < num_cols; j++) {
-        ret = kastore_gets(
-            store, read_cols[j].name, read_cols[j].array_dest, &len, &type);
-        if (ret != 0) {
-            ret = tsk_set_kas_error(ret);
-            goto out;
-        }
-        last_len = *read_cols[j].len_dest;
-        if (last_len == (tsk_size_t) -1) {
-            *read_cols[j].len_dest = (tsk_size_t)(len - read_cols[j].len_offset);
-        } else if ((last_len + read_cols[j].len_offset) != (tsk_size_t) len) {
-            ret = TSK_ERR_FILE_FORMAT;
-            goto out;
-        }
-        if (type != read_cols[j].type) {
-            ret = TSK_ERR_FILE_FORMAT;
+        if (kastore_containss(store, read_cols[j].name)) {
+            ret = kastore_gets(
+                store, read_cols[j].name, read_cols[j].array_dest, &len, &type);
+            if (ret != 0) {
+                ret = tsk_set_kas_error(ret);
+                goto out;
+            }
+            last_len = *read_cols[j].len_dest;
+            if (last_len == (tsk_size_t) -1) {
+                *read_cols[j].len_dest = (tsk_size_t)(len - read_cols[j].len_offset);
+            } else if ((last_len + read_cols[j].len_offset) != (tsk_size_t) len) {
+                ret = TSK_ERR_FILE_FORMAT;
+                goto out;
+            }
+            if (type != read_cols[j].type) {
+                ret = TSK_ERR_FILE_FORMAT;
+                goto out;
+            }
+        } else if (!(read_cols[j].options && TSK_COL_OPTIONAL)) {
```

Comment:
Did you mean boolean && or bitwise &? Probably bitwise?

Reply:
Opps, fixed in https://github.com/tskit-dev/tskit/pull/496/commits/ace3d7b7de5568a04a00b4a2ff76b42f138983c4